File: backend/src/Liturgy.Domain/Decision.cs
namespace Liturgy.Domain;

public class Decision
{
    public Guid Id { get; set; }
    public Guid ProjectId { get; set; }
    public DiscernmentPath ChosenPath { get; set; }
    public string Rationale { get; set; } = string.Empty;
    public string PrayedOverWith { get; set; } = string.Empty;
    public DateTimeOffset DecidedAt { get; set; }
}

File: frontend/projects/liturgy/api/src/lib/models/decision.ts
import { DiscernmentPath } from './discernment-path';

/** A project's Discern decision: the chosen path plus the recorded rationale. */
export interface Decision {
  chosenPath: DiscernmentPath | null;
  rationale: string;
  prayedOverWith: string;
  decidedAt: string | null;
}

/** Payload for recording/updating a Discern decision. */
export interface SaveDecisionRequest {
  chosenPath: DiscernmentPath;
  rationale: string;
  prayedOverWith: string;
}
